pipeline_reviews/load.py

```diff
@@ -0,0 +1,54 @@
+"""Loads reviews into the database"""
+
+import pandas as pd
+from pandas import DataFrame
+from psycopg2 import Error
+from psycopg2.extensions import connection
+from psycopg2.extras import execute_batch
+
+from extract import get_db_connection
+from transform import remove_unnamed, remove_empty_rows
+
+
+def get_game_ids_foreign_key_values(reviews_df: DataFrame) -> DataFrame:
+    """Returns data-frame with game_ids from db for
+    foreign keys"""
+    conn = get_db_connection()
+    cache_dict = dict()
+    reviews_df["game_id"] = reviews_df["game_id"].apply(
+        lambda row: get_game_ids(conn, row, cache_dict))
+    reviews_df = remove_empty_rows(reviews_df)
+    return reviews_df
+
+
+def get_game_ids(conn: connection, app_id: int, cache: dict) -> int | None:
+    """Returns game_id from game table from db (foreign key)"""
+    if str(app_id) in cache.keys():
+        return cache[str(app_id)]
+    try:
+        with conn.cursor() as cur:
+            cur.execute("""SELECT game_id FROM game WHERE app_id = %s""", app_id)
+            game_id = cur.fetchone()
+            conn.close()
+            game_id = game_id[["game_id"]]
+            cache[str(app_id)] = game_id
```

Comment:
close cursor here

Reply:
So, connection needs closing as it was not entered and closed but cursor here was. So this is not needed here due to the "with" statement. If you ever mocked it, you could have noticed that you need to do .__enter__ well this is for cursor to be opened but it also does __exit__ (this you might have not seen with mocking but if you ever created mocking classes for it, you'd have to make one of this too as it does close the cursor). It does however not close the connection :) but this is added